--- setup.py.orig	2016-01-29 10:36:41.000000000 -0500
+++ setup.py	2016-05-11 19:04:51.000000000 -0400
@@ -53,6 +53,9 @@
 ISRELEASED = True
 VERSION = '%d.%d.%d' % (MAJOR, MINOR, MICRO)
 
+static_fftw_path = os.environ.get('STATIC_FFTW_DIR', None)
+link_static_fftw = static_fftw_path is not None
+
 def get_package_data():
     from pkg_resources import get_build_platform
 
@@ -121,8 +124,27 @@
         
         have_cython = False
 
+    if link_static_fftw:
+        from pkg_resources import get_build_platform
+        if get_build_platform() in ('win32', 'win-amd64'):
+            lib_pre = ''
+            lib_ext = '.lib'
+        else:
+            lib_pre = 'lib'
+            lib_ext = '.a'
+        extra_link_args = []
+        for lib in common_extension_args['libraries']:
+            extra_link_args.append(
+                os.path.join(static_fftw_path, lib_pre + lib + lib_ext))
+        # now that full paths to libraries are in extra_link_args remove them
+        # from common_extension_args
+        common_extension_args['libraries'] = []
+    else:
+        extra_link_args = []
+
     ext_modules = [
         Extension('pyfftw.pyfftw', sources=sources, 
+                  extra_link_args=extra_link_args,
                   **common_extension_args)]
 
     if have_cython:
